pkg/services/baremetal/host/host.go

```diff
@@ -608,13 +612,22 @@ func (s *Service) actionRegistering(_ context.Context) actionResult {
 			return actionError{err: fmt.Errorf("failed to handle incomplete boot - registering: %w", err)}
 		}
 
-		failed, err := s.handleIncompleteBoot(true, isSSHTimeoutError, isSSHConnectionRefusedError)
+		failed, err := s.handleIncompleteBoot(ctx, true, isSSHTimeoutError, isSSHConnectionRefusedError)
 		if failed {
 			return s.recordActionFailure(infrav1.PermanentError, err.Error())
 		}
 		if err != nil {
 			return actionError{err: fmt.Errorf(errMsgFailedHandlingIncompleteBoot, err)}
 		}
+
+		if !isSSHTimeoutError && !isSSHConnectionRefusedError {
```

Comment:
are you sure about this code block? We do handle these cases in the handleIncompleteBoot function. Also this case here. We handle it in a way that says "boot didn't happen so we trigger another one". If you want to update this logic, then you should change the handleIncompleteBoot function IMO

Reply:
I would love to update that, so that it is easier to understand. But not in this PR, and not now.

Thank you for your hint. I removed that.